java/HealthVault Mobile/HVSample/src/main/java/com/microsoft/hsg/android/hvsample/WeightActivity.java

```diff
@@ -0,0 +1,163 @@
+package com.microsoft.hsg.android.hvsample;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import com.microsoft.hsg.HVException;
+import com.microsoft.hsg.android.simplexml.HealthVaultApp;
+import com.microsoft.hsg.android.simplexml.ShellActivity;
+import com.microsoft.hsg.android.simplexml.client.HealthVaultClient;
+import com.microsoft.hsg.android.simplexml.client.RequestCallback;
+import com.microsoft.hsg.android.simplexml.methods.getthings3.request.ThingRequestGroup2;
+import com.microsoft.hsg.android.simplexml.methods.getthings3.response.ThingResponseGroup2;
+import com.microsoft.hsg.android.simplexml.things.thing.Thing2;
+import com.microsoft.hsg.android.simplexml.things.types.types.PersonInfo;
+import com.microsoft.hsg.android.simplexml.things.types.types.Record;
+import com.microsoft.hsg.android.simplexml.things.types.weight.Weight;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.view.Window;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ListView;
+import android.widget.Spinner;
+import android.widget.TextView;
+import android.widget.Toast;
+import android.widget.AdapterView.OnItemSelectedListener;
+
+public class WeightActivity extends Activity {
+
+	private HealthVaultApp mService;
+    private HealthVaultClient mHVClient;
+    private Record mCurrentRecord;
+    
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.weight);
+        mService = HealthVaultApp.getInstance();
+        mHVClient = new HealthVaultClient();
+        
+        Button weightsBtn = (Button) findViewById(R.id.addWeight);
+        final EditText editText = (EditText) findViewById(R.id.weightInput);
+		
+        weightsBtn.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View view) {
+            	if (mService.isAppConnected()) {
+                    try {
+                        putWeight(editText.getText().toString());
+                        editText.setText("");
+                    }
+                    catch (Exception e) {
+                        Toast.makeText(WeightActivity.this, "Please enter a weight!", Toast.LENGTH_SHORT).show();
+                    }
+            	}
+            }
+        });
+
+        setTitle("Weight Sample");
+    }
+    
+	@Override
+	protected void onStart() {
+		super.onStart();
+        mHVClient.start();
+	}
+    
+    @Override
+    protected void onResume()
+    {
+        super.onResume();
+        mCurrentRecord = HealthVaultApp.getInstance().getCurrentRecord();
```

Comment:
Yes, that is correct.  The activity is not visible and need to get the data on the foreground/on resume

---
In reply to: [112605421](https://github.com/Microsoft/healthvault-samples/pull/4#discussion_r112605421) [](ancestors = 112605421)